### PR TITLE
capistrano configs and shared-configs

### LINKED ISSUE #33: Convert to using shared_configs
TODO:
- [x] shared_configs are in:
  - https://github.com/sul-dlss/shared_configs/tree/sdr-preservation-core_dev
  - https://github.com/sul-dlss/shared_configs/tree/sdr-preservation-core_stage
  - https://github.com/sul-dlss/shared_configs/tree/sdr-preservation-core_prod
- [x] puppet deployment for shared_configs
  - @eefahy recommends that we don't hack puppet for this because it's not designed to work well with two apps deployed to the same system
  - add tasks to the capistrano-shared-configs gem:
    - https://github.com/sul-dlss/capistrano-shared_configs/pull/5
- [x] the sdr-preservation-core should contain capistrano deploy configs
  - [x] proposed changes are in PR #58 
- [x] the sdr-configs to be purged of any sdr-preservation-core configs
  - [x] proposed changes are in PR https://github.com/sul-dlss/sdr-configs/pull/6

Related issues
- https://github.com/sul-dlss/sdr-configs/issues/1

Blockers:
- [x] #30 requires bundle updates to resolve security vulnerabilities (capistrano uses bundle-audit and it fails because master branch has vulnerabilities), but this is complicated by various DLSS dependencies and working through these updates has already been done on the `develop` branch, but the current deployment systems are still running the `master` branch.  The work to merge `develop` into `master` is complicated because simple cherry-picks are often not working and there are conflicts between these branches - this work is tracked in various PRs that are trying to resolve conflicts in #34 and hopefully get #34 merged.
- [x] blockers should be resolved by #57 and #55

### PR DESCRIPTION
**Pull request by dazza-codes**
Partial solution for #33 - it works and it's deployed on `sul-sdr-services-dev.stanford.edu` along with the current shared_configs for that system, which seems to be OK in rescue, see
- https://sul-lyberservices-dev.stanford.edu/robots/workers/sul-sdr-services-dev.stanford.edu

e.g.

```
$ bundle exec cap dev deploy:check
...
00:00 git:wrapper
      01 mkdir -p /tmp
    ✔ 01 sdr2service@sul-sdr-services-dev.stanford.edu 1.213s
      Uploading /tmp/git-ssh-sdr-preservation-core-dev-dlweber.sh 100.0%
      02 chmod 700 /tmp/git-ssh-sdr-preservation-core-dev-dlweber.sh
    ✔ 02 sdr2service@sul-sdr-services-dev.stanford.edu 1.066s
Please enter branch (capistrano-shared-configs-33): 
00:05 git:check
      01 git ls-remote https://github.com/sul-dlss/sdr-preservation-core.git HEAD
      01 844d6f93a6f8185f51e4e1a2d79c4df7171050b3	HEAD
    ✔ 01 sdr2service@sul-sdr-services-dev.stanford.edu 1.868s
00:06 deploy:check:directories
      01 mkdir -p /var/sdr2service/sdr-preservation-core/shared /var/sdr2service/sdr-preservation-core/releases
    ✔ 01 sdr2service@sul-sdr-services-dev.stanford.edu 1.145s
00:08 deploy:check:linked_dirs
      01 mkdir -p /var/sdr2service/sdr-preservation-core/shared/.binstubs /var/sdr2service/sdr-preservation-core/shared/log /var/sdr2service/sdr-preservation-core/shared/run /var/sdr2service/sdr-…
    ✔ 01 sdr2service@sul-sdr-services-dev.stanford.edu 1.145s
```

### CODE REVIEW

**Comment by darrenleeweber** [comment]
Also works on `stage`, e.g.
```
bundle exec cap stage deploy:check
Pseudo-terminal will not be allocated because stdin is not a terminal.
Pseudo-terminal will not be allocated because stdin is not a terminal.
The authenticity of host 'sdr-services-test2.stanford.edu (172.20.207.30)' can't be established.
RSA key fingerprint is SHA256:crFySdqC1FoWiSWDLrobxSg0AkWwD/bxjQOlDiFS8L0.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'sdr-services-test2.stanford.edu,172.20.207.30' (RSA) to the list of known hosts.
rvm 1.25.28 (stable) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]
rvm 1.26.3 (latest) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]
ruby-2.1.2
ruby-2.1.2
ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-linux]
ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-linux]
00:00 git:wrapper
      01 mkdir -p /tmp
    ✔ 01 sdr2service@sdr-services-test.stanford.edu 1.067s
    ✔ 01 sdr2service@sdr-services-test2.stanford.edu 1.589s
      Uploading /tmp/git-ssh-sdr-preservation-core-stage-dlweber.sh 100.0%
      02 chmod 700 /tmp/git-ssh-sdr-preservation-core-stage-dlweber.sh
      Uploading /tmp/git-ssh-sdr-preservation-core-stage-dlweber.sh 100.0%
    ✔ 02 sdr2service@sdr-services-test.stanford.edu 0.986s
    ✔ 02 sdr2service@sdr-services-test2.stanford.edu 1.340s
Please enter branch (capistrano-shared-configs-33): 
00:06 git:check
      01 git ls-remote https://github.com/sul-dlss/sdr-preservation-core.git HEAD
      01 844d6f93a6f8185f51e4e1a2d79c4df7171050b3	HEAD
    ✔ 01 sdr2service@sdr-services-test.stanford.edu 1.445s
      01 844d6f93a6f8185f51e4e1a2d79c4df7171050b3	HEAD
    ✔ 01 sdr2service@sdr-services-test2.stanford.edu 2.356s
00:08 deploy:check:directories
      01 mkdir -p /var/sdr2service/sdr-preservation-core/shared /var/sdr2service/sdr-preservation-core/releases
    ✔ 01 sdr2service@sdr-services-test.stanford.edu 1.069s
    ✔ 01 sdr2service@sdr-services-test2.stanford.edu 1.470s
00:10 deploy:check:linked_dirs
      01 mkdir -p /var/sdr2service/sdr-preservation-core/shared/.binstubs /var/sdr2service/sdr-preservation-core/shared/log /var/sdr2service/sdr-preservation-core/shared/run /var/sdr2service/sdr-…
    ✔ 01 sdr2service@sdr-services-test.stanford.edu 1.088s
    ✔ 01 sdr2service@sdr-services-test2.stanford.edu 1.431s
Pseudo-terminal will not be allocated because stdin is not a terminal.
Pseudo-terminal will not be allocated because stdin is not a terminal.
```

File: config/deploy.rb [hunk 1]
```diff
@@ -29,12 +22,26 @@
...
+set :bundle_without, 'development test'
+set :bundle_flags, '--deployment'
+
 # Default value for :linked_files is []
 # set :linked_files, %w{config/database.yml}
 
 # Default value for linked_dirs is []
 # set :linked_dirs, %w{bin log tmp/pids tmp/cache tmp/sockets vendor/bundle public/system}
-set :linked_dirs, %w(log run config/environments config/certs)
+set :linked_dirs, %w(.binstubs log run config/environments config/certs)
```
Comment (jcoyne) [inline config/deploy.rb]: Why is .binstubs a linked directory?

**Review by jcoyne — COMMENTED** [review]
Looks good. I just don't understand how this is using .binstubs

File: config/deploy.rb (hunk 1 above)
Comment (darrenleeweber) [inline config/deploy.rb]: The bundle install creates this path as a utility and linking it here enables easy use of the command utils in there, http://bundler.io/v1.10/bundle_binstubs.html

Comment (jcoyne) [inline config/deploy.rb]: Why do we want to use binstubs on the server?  What were we doing prior to this change?  Is this something that needs to be part of this PR?

Comment (darrenleeweber) [inline config/deploy.rb]: What's wrong with it?

Comment (cbeer) [inline config/deploy.rb]: What's wrong with the default behavior?

Comment (darrenleeweber) [inline config/deploy.rb]: lol, this question tennis is not getting us anywhere; what's the objection to using .binstubs?  Is it a blocker on this PR?  Are you asking me to remove it or this PR is blocked?  What do you want?

Comment (jcoyne) [inline config/deploy.rb]: I think moving this to a different PR would be best.  This isn't required to achieve the primary goal of the PR.

Comment (jcoyne) [inline config/deploy.rb]: Furthermore, it's suggested that binstubs be checked in if you're going to use them: https://github.com/rbenv/rbenv/wiki/Understanding-binstubs#bundler-generated-binstubs
> You are encouraged to check these binstubs in the project's version control so your colleagues might benefit from them.

If they're linked directories, that implies they aren't in version control.

Comment (darrenleeweber) [inline config/deploy.rb]: OK, I'll try to separate this out (and better check there are no cron or other things that depend on having access to this bin path).

Comment (darrenleeweber) [inline config/deploy.rb]: The default is `set :bundle_binstubs, -> { shared_path.join('bin') }` and this project already has a `bin` directory, so hopefully we have no bin-file name conflicts (probably not).

**Review by jcoyne — APPROVED** [review]
